lib/github/components/FileCommits/index.js

```diff
@@ -0,0 +1,152 @@
+import { Link } from '../../../../server/routes'
+import { gql, graphql } from 'react-apollo'
+import { css } from 'glamor'
+import { swissTime } from '../../../utils/format'
+
+import { colors } from '@project-r/styleguide'
+
+const timeFormat = swissTime.format('%d. %B %Y, %H:%M Uhr')
+
+const styles = {
+  commits: css({
+    listStyleType: 'none',
+    margin: 0,
+    maxHeight: '300px',
+    overflow: 'scroll',
+    padding: 0
+  }),
+  commit: css({
+    backgroundColor: colors.secondaryBg,
+    borderBottom: `1px solid ${colors.divider}`,
+    fontSize: '13px',
+    padding: '5px',
+    position: 'relative'
+  }),
+  date: css({
+    display: 'block',
+    fontSize: '11px'
+  })
+}
+
+const query = gql`
+  query tree($login: String!, $repository: String!, $filename: String!) {
+    repository(owner: $login, name: $repository) {
+      name
+      owner {
+        id
+        login
+      }
+      refs(refPrefix: "refs/heads/", first: 100) {
+        nodes {
+          name
+          target {
+            ... on Commit {
+              oid
+              history(first: 30, path: $filename) {
+                nodes {
+                  author {
+                    email
+                    name
+                  }
+                  committedDate
+                  oid
+                  messageHeadline
+                  messageBody
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+`
+
+const FileCommits = ({ data: { loading, error, repository }, path }) => {
+  if (loading) {
+    return <div>Loading</div>
+  }
+
+  let branches
+  let allCommits = []
+
+  try {
+    branches = repository.refs.nodes
+    let oids = {}
+    for (let i = 0; i < branches.length; i++) {
```

Comment:
I would recommend a for-of loop, because it's more readable IMHO and very 2017ish
for (let branch of branches)